### PR TITLE
feat: use environment variable to custom path prefix

### PR DESCRIPTION
**Pull request by kecrily**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fix [zh-hans.docs.eslint.org](https://zh-hans.docs.eslint.org) style lost.

### CODE REVIEW

File: docs/.eleventy.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ module.exports = function(eleventyConfig) {
 
     let pathPrefix = "/docs/head/";
 
-    if (process.env.CONTEXT === "deploy-preview") {
+    if (process.env.CONTEXT === "deploy-preview" || process.env.BRANCH === "main") {
         pathPrefix = "/";
```
Comment (mdjermanovic) [inline docs/.eleventy.js:38]: I think this would break all pages under https://eslint.org/docs/head/

Comment (kecrily) [reply docs/.eleventy.js:38]: You're right. The change is wrong, I missed something.

I took a closer look, and now I have a problem. zh-hans site uses the exact same code as this, but they both have different resource prefixes.

The root site's resources are under `/docs/head/`, while the zh-hans site's resources are under `/`. Both of them are `HEAD` version. Is there a detail I'm missing?

Comment (mdjermanovic) [reply docs/.eleventy.js:38]: Maybe we could use an environment variable that would override this logic?

**Comment by nzakas** [comment]
This is already working as intended. [zh-hans.docs.eslint.org](https://zh-hans.docs.eslint.org/) Is intended to be viewed through zh-hans.eslint.org/docs just like https://docs-eslint.netlify.app/ is intended to be viewed from eslint.org/docs.

Now that the Chinese docs are live, we can enable them through zh-hans.eslint.org:
https://github.com/eslint/eslint.org/pull/357

**Comment by kecrily** [comment]
I've noticed that visiting docs.eslint.org directly doesn't result in a missing stylesheet. Instead, en-hans.docs.eslint.org has

**Comment by kecrily** [comment]
I've noticed that visiting docs.eslint.org directly doesn't result in missing stylesheet. Instead, zh-hans.docs.eslint.org has

**Pull request closed without merging**